src/processor/execution_context.cpp

```diff
@@ -0,0 +1,61 @@
+#include "processor/execution_context.h"
+
+#include "processor/operator/persistent/reader/csv/csv_error.h"
+#include "processor/result/factorized_table_util.h"
+#include "processor/warning_schema.h"
+
+namespace kuzu {
+using namespace common;
+namespace processor {
+
+ExecutionContext::ExecutionContext(common::Profiler* profiler, main::ClientContext* clientContext,
+    uint64_t queryID)
+    : queryID{queryID}, profiler{profiler}, clientContext{clientContext} {
+    FactorizedTableSchema tableSchema;
+
+    const auto dataTypes = WarningSchema::getColumnDataTypes();
+    for (idx_t i = 0; i < WarningSchema::getNumColumns(); ++i) {
+        tableSchema.appendColumn(
+            ColumnSchema{false, i, LogicalTypeUtils::getRowLayoutSize(dataTypes[i])});
+    }
+
+    warningTable = std::make_unique<FactorizedTable>(this->clientContext->getMemoryManager(),
+        std::move(tableSchema));
+}
+
+// marking this function with const would be misleading here
+// NOLINTNEXTLINE(readability-make-member-function-const)
+void ExecutionContext::setWarningMessages(const std::vector<PopulatedCSVError>& messages,
+    uint64_t messageLimit) {
+    common::UniqLock lock{mtx};
+    warningLimit = messageLimit;
```

Comment:
I'm a bit confused on this interface.
1. Why is `warningLimit` set here? Shouldn't it be set from config?
2. It makes more sense to me if this interface is `appendWarningMessages`, instead of `setWarningMessages`.
3. I don't see the logic handling the case that num of warnings exceed max limit, maybe I'm missing that part somewhere else, I expect we throw errors. Or we currently ignore warnings exceeding limit?

Reply:
I also forgot to mention that currently if we go over the warning limit it will print `4096+ warnings` if the warning limit is 4096 and not the actual number of warnings let me know if we want to change this.

Reply:
Update: I removed warning limit since it isn't used anymore. As I said before the warning limiting is currently handled in the CSV error handler.